src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -195,6 +195,12 @@ public Task<IActionResult> ExecuteAsync(StoredProcedureRequestContext context, s
             throw new NotImplementedException();
         }
 
+        /// <inheritdoc />
+        public Task<JsonDocument> ExecuteAsyncAndGetResponseAsJsonDocument(FindRequestContext context, string datasourceName = "")
+        {
+            throw new NotImplementedException();
```

Comment:
The default implementation can be provided for the method in the interface `IQueryEngine`.